libra-node/src/main_node.rs

```diff
@@ -351,22 +359,16 @@ pub fn setup_environment(node_config: &mut NodeConfig) -> LibraHandle {
     );
     debug!("Mempool started in {} ms", instant.elapsed().as_millis());
 
-    if let Some((peer_id, runtime, mut network_builder)) = validator_network_provider {
-        // Note: We need to start network provider before consensus, because the consensus
-        // initialization is blocked on state synchronizer to sync to the initial root ledger
-        // info, which in turn cannot make progress before network initialization
-        // because the NewPeer events which state synchronizer uses to know its
-        // peers are delivered by network provider. If we were to start network
-        // provider after consensus, we create a cyclic dependency from
-        // network provider -> consensus -> state synchronizer -> network provider. This deadlock
-        // was observed in GitHub Issue #749. A long term fix might be make
-        // consensus initialization async instead of blocking on state synchronizer.
-        let (consensus_network_sender, consensus_network_events) =
-            consensus::network_interface::add_to_network(&mut network_builder);
-        let _listen_addr = network_builder.build();
-        network_runtimes.push(runtime);
-        debug!("Network started for peer_id: {}", peer_id);
-
+    // Note: We need to start network provider before consensus, because the consensus
```

Comment:
haha, you don't want to touch this overly verbose statement that explicitly boils down to

Start state sync first to prepare the ledger for consensus and prevent a cyclic dependency that might cause a deadlock. See GitHub Issue #749 [link].

Reply:
I'm not touching it.  I'm leaving it in place.   problem is that the network start code all moved up :)